internal/docstore/driver/driver.go

```diff
@@ -71,3 +83,28 @@ type Mod struct {
 	FieldPath []string
 	Value     interface{}
 }
+
+// An ActionListError contains all the errors encountered from a call to RunActions,
+// and the positions of the corresponding actions.
+type ActionListError []struct {
```

Comment:
Consider making a type for the struct so that it doesn't have to be inlined everywhere?

Reply:
We inline it once. Giving it a name would pollute the API forever at the cost of a tiny bit of implementation pain.

Reply:
If we find that implementing the unordered mode means writing that struct out a lot, in every driver, then we can just have an unexported function that returns a struct value.